--- a/build.zig
+++ b/build.zig
@@ -76,42 +76,6 @@
         "Update translation files",
     );
 
-    // Ghostty resources like terminfo, shell integration, themes, etc.
-    const resources = try buildpkg.GhosttyResources.init(b, &config, &deps);
-    const i18n = if (config.i18n) try buildpkg.GhosttyI18n.init(b, &config) else null;
-
-    // Ghostty executable, the actual runnable Ghostty program.
-    const exe = try buildpkg.GhosttyExe.init(b, &config, &deps);
-
-    // Ghostty docs
-    const docs = try buildpkg.GhosttyDocs.init(b, &deps);
-    if (config.emit_docs) {
-        docs.install();
-    } else if (config.target.result.os.tag.isDarwin()) {
-        // If we aren't emitting docs we need to emit a placeholder so
-        // our macOS xcodeproject builds since it expects the `share/man`
-        // directory to exist to copy into the app bundle.
-        docs.installDummy(b.getInstallStep());
-    }
-
-    // Ghostty webdata
-    const webdata = try buildpkg.GhosttyWebdata.init(b, &deps);
-    if (config.emit_webdata) webdata.install();
-
-    // Ghostty bench tools
-    const bench = try buildpkg.GhosttyBench.init(b, &deps);
-    if (config.emit_bench) bench.install();
-
-    // Ghostty dist tarball
-    const dist = try buildpkg.GhosttyDist.init(b, &config);
-    {
-        const step = b.step("dist", "Build the dist tarball");
-        step.dependOn(dist.install_step);
-        const check_step = b.step("distcheck", "Install and validate the dist tarball");
-        check_step.dependOn(dist.check_step);
-        check_step.dependOn(dist.install_step);
-    }
-
     // libghostty-vt
     const libghostty_vt_shared = shared: {
         if (config.target.result.cpu.arch.isWasm()) {
@@ -154,6 +118,55 @@
     // Helpgen
     if (config.emit_helpgen) deps.help_strings.install();
 
+
+    if (config.emit_lib_vt and
+        config.app_runtime == .none and
+        !config.emit_exe and
+        !config.emit_bench and
+        !config.emit_docs and
+        !config.emit_macos_app and
+        !config.emit_webdata and
+        !config.emit_xcframework)
+    {
+        return;
+    }
+
+    // Ghostty resources like terminfo, shell integration, themes, etc.
+    const resources = try buildpkg.GhosttyResources.init(b, &config, &deps);
+    const i18n = if (config.i18n) try buildpkg.GhosttyI18n.init(b, &config) else null;
+
+    // Ghostty executable, the actual runnable Ghostty program.
+    const exe = try buildpkg.GhosttyExe.init(b, &config, &deps);
+
+    // Ghostty docs
+    const docs = try buildpkg.GhosttyDocs.init(b, &deps);
+    if (config.emit_docs) {
+        docs.install();
+    } else if (config.target.result.os.tag.isDarwin()) {
+        // If we aren't emitting docs we need to emit a placeholder so
+        // our macOS xcodeproject builds since it expects the `share/man`
+        // directory to exist to copy into the app bundle.
+        docs.installDummy(b.getInstallStep());
+    }
+
+    // Ghostty webdata
+    const webdata = try buildpkg.GhosttyWebdata.init(b, &deps);
+    if (config.emit_webdata) webdata.install();
+
+    // Ghostty bench tools
+    const bench = try buildpkg.GhosttyBench.init(b, &deps);
+    if (config.emit_bench) bench.install();
+
+    // Ghostty dist tarball
+    const dist = try buildpkg.GhosttyDist.init(b, &config);
+    {
+        const step = b.step("dist", "Build the dist tarball");
+        step.dependOn(dist.install_step);
+        const check_step = b.step("distcheck", "Install and validate the dist tarball");
+        check_step.dependOn(dist.check_step);
+        check_step.dependOn(dist.install_step);
+    }
+
     // Runtime "none" is libghostty, anything else is an executable.
     if (config.app_runtime != .none) {
         if (config.emit_exe) {
